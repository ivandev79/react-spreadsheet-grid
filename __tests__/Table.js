import React from 'react';
import renderer from 'react-test-renderer';
import Table from './../src/table';
import { columnsWithTextValues,
    columnsWithComponents,
    rows } from './../testKit'

test('Table without rows', () => {
    const tree = renderer.create(
        <Table
            columns={columnsWithTextValues}
            getRowKey={row => row.id}
            headerHeight={50}
            cellHeight={50}
            first={0}
            last={2}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('Table with columns with text values', () => {
    const tree = renderer.create(
        <Table
            columns={columnsWithTextValues}
            rows={rows}
            getRowKey={row => row.id}
            headerHeight={50}
            cellHeight={50}
            first={0}
            last={2}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('Table with columns with components', () => {
    const tree = renderer.create(
        <Table
            columns={columnsWithComponents}
            rows={rows}
            getRowKey={row => row.id}
            headerHeight={50}
            cellHeight={50}
            first={0}
            last={2}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

