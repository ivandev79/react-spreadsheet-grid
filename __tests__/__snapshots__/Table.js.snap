// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table with columns with components 1`] = `
<table
  className="SpreadsheetTable"
  style={
    Object {
      "top": "NaNpx",
    }
  }
>
  <thead
    style={
      Object {
        "display": "",
        "height": "50px",
      }
    }
  >
    <tr>
      <th>
        Name
      </th>
      <th>
        Second name
      </th>
      <th>
        Position
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      style={
        Object {
          "height": 50,
        }
      }
    >
      <td
        className="SpreadsheetTable__cell"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        Name 1
      </td>
      <td
        className="SpreadsheetTable__cell"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        <div
          className="SpreadsheetTableDropdown"
        >
          <div
            onClick={[Function]}
          >
            <div
              className="SpreadsheetTableSelectHeader"
            />
          </div>
          <div
            className="SpreadsheetTableDropdown__body"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <div>
              <div
                className="SpreadsheetTableSelectItem"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              />
            </div>
          </div>
        </div>
      </td>
      <td
        className="SpreadsheetTable__cell"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        <input
          className="SpreadsheetTable__Input"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder={undefined}
          value="Position 1"
        />
      </td>
    </tr>
    <tr
      style={
        Object {
          "height": 50,
        }
      }
    >
      <td
        className="SpreadsheetTable__cell"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        Name 2
      </td>
      <td
        className="SpreadsheetTable__cell"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        <div
          className="SpreadsheetTableDropdown"
        >
          <div
            onClick={[Function]}
          >
            <div
              className="SpreadsheetTableSelectHeader"
            />
          </div>
          <div
            className="SpreadsheetTableDropdown__body"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <div>
              <div
                className="SpreadsheetTableSelectItem"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              />
            </div>
          </div>
        </div>
      </td>
      <td
        className="SpreadsheetTable__cell"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        <input
          className="SpreadsheetTable__Input"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder={undefined}
          value="Position 2"
        />
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table with columns with text values 1`] = `
<table
  className="SpreadsheetTable"
  style={
    Object {
      "top": "NaNpx",
    }
  }
>
  <thead
    style={
      Object {
        "display": "",
        "height": "50px",
      }
    }
  >
    <tr>
      <th>
        Name
      </th>
      <th>
        Second name
      </th>
      <th>
        Position
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      style={
        Object {
          "height": 50,
        }
      }
    >
      <td
        className="SpreadsheetTable__cell"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        Name 1
      </td>
      <td
        className="SpreadsheetTable__cell"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        Second name 1
      </td>
      <td
        className="SpreadsheetTable__cell"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        Position 1
      </td>
    </tr>
    <tr
      style={
        Object {
          "height": 50,
        }
      }
    >
      <td
        className="SpreadsheetTable__cell"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        Name 2
      </td>
      <td
        className="SpreadsheetTable__cell"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        Second name 2
      </td>
      <td
        className="SpreadsheetTable__cell"
        onClick={[Function]}
        onDoubleClick={[Function]}
      >
        Position 2
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table without rows 1`] = `
<table
  className="SpreadsheetTable"
  style={
    Object {
      "top": "NaNpx",
    }
  }
>
  <thead
    style={
      Object {
        "display": "",
        "height": "50px",
      }
    }
  >
    <tr>
      <th>
        Name
      </th>
      <th>
        Second name
      </th>
      <th>
        Position
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="SpreadsheetTable__placeholder"
    >
      <td
        colSpan={3}
      >
        <div />
      </td>
    </tr>
  </tbody>
</table>
`;
